### PR TITLE
Template update for nf-core/tools version 3.0.2

### PR DESCRIPTION
**Pull request by LilyAnderssonLee**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/genomic-medicine-sweden/meta-val/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the genomic-medicine-sweden/meta-val _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,4 @@ results/
 testing/
 testing*
 *.pyc
+null/
```
Comment (sofstam) [inline .gitignore:9]: What is this folder?

File: assets/schema_input.json [hunk 2]
```diff
@@ -8,29 +8,25 @@
         "type": "object",
         "properties": {
             "sample": {
-                "type": "string",
-                "exists": true,
+                "type": ["string", "integer"],
```
Comment (sofstam) [inline assets/schema_input.json:11]: Isn't the `exists` necessary?

File: assets/schema_input.json [hunk 3]
```diff
@@ -51,15 +47,13 @@
...
-                "dependentRequired": ["kraken2_report"]
+                "errorMessage": "Kraken2 classification results must be provided."
             },
             "kraken2_taxpasta": {
                 "type": "string",
                 "format": "file-path",
                 "pattern": "^\\S+\\.tsv$",
-                "errorMessage": "Standardized kraken2 profiling with taxonomic information obtained from taxpasta.",
-                "dependentRequired": ["kraken2_report", "kraken2_result"]
+                "errorMessage": "Standardized kraken2 profiling with taxonomic information obtained from taxpasta."
```
Comment (sofstam) [inline assets/schema_input.json:61]: Same about the `dependentRequired`

File: subworkflows/local/utils_nfcore_metaval_pipeline/main.nf [hunk 4]
```diff
@@ -54,16 +50,10 @@ workflow PIPELINE_INITIALISATION {
...
-    UTILS_NFVALIDATION_PLUGIN (
-        help,
-        workflow_command,
-        pre_help_text,
-        post_help_text,
+    UTILS_NFSCHEMA_PLUGIN (
+        workflow,
         validate_params,
-        "nextflow_schema.json"
+        null
```
Comment (sofstam) [inline subworkflows/local/utils_nfcore_metaval_pipeline/main.nf:56]: What is the purpose of `null` here?

File: workflows/metaval.nf [hunk 5]
```diff
@@ -124,48 +123,33 @@ workflow METAVAL {
...
+    ch_multiqc_files                      = ch_multiqc_files.mix(ch_collated_versions)
+    ch_multiqc_files                      = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml',sort: true))
 
     MULTIQC (
         ch_multiqc_files.collect(),
         ch_multiqc_config.toList(),
         ch_multiqc_custom_config.toList(),
-        ch_multiqc_logo.toList()
+        ch_multiqc_logo.toList(),
+        [],
```
Comment (sofstam) [inline workflows/metaval.nf:151]: Same question here about the empty lists.

**Review by sofstam — COMMENTED** [review]
Nice work! A few questions!

File: .gitignore (hunk 1 above)
Comment (LilyAnderssonLee) [reply .gitignore:9]: A `null` folder sometimes generated from a failed analysis, so git will ignore checking o this folder.

File: assets/schema_input.json (hunk 2 above)
Comment (LilyAnderssonLee) [reply assets/schema_input.json:11]: nextflow schema is migrating to the new version, so `exists` is only used for input checks now in the Nextflow schema. I’ll ping you in the Slack channel of details.

File: assets/schema_input.json (hunk 3 above)
Comment (LilyAnderssonLee) [reply assets/schema_input.json:61]: `dependentRequired` is still included in the schema, just moved to the bottom with the new format.

File: subworkflows/local/utils_nfcore_metaval_pipeline/main.nf (hunk 4 above)
Comment (LilyAnderssonLee) [reply subworkflows/local/utils_nfcore_metaval_pipeline/main.nf:56]: From the template.

File: workflows/metaval.nf (hunk 5 above)
Comment (LilyAnderssonLee) [reply workflows/metaval.nf:151]: The new MULTIQC module needs 6 channles. Here an empty list stands for `path(sample_names)` which we usually don't define it.

File: assets/schema_input.json (hunk 2 above)
Comment (LilyAnderssonLee) [reply assets/schema_input.json:11]: Some updates. `"exists": true` only for file path check.

File: assets/schema_input.json [hunk 6]
```diff
@@ -37,64 +34,76 @@
...
         },
-        "required": ["sample", "run_accession", "instrument_platform", "fastq_1"]
-    }
+        "required": ["sample", "run_accession", "instrument_platform", "fastq_1"],
+        "dependentRequired": {
+            "kraken2_result": ["kraken2_report"],
+            "kraken2_taxpasta": ["kraken2_report", "kraken2_result"],
+            "centrifuge_result": ["centrifuge_report"],
+            "centrifuge_taxpasta": ["centrifuge_report", "centrifuge_result"],
+            "diamond_taxpasta": ["diamond"]
```
Comment (sofstam) [inline assets/schema_input.json:105]: Don't we need here `diamond_result` too?

Comment (LilyAnderssonLee) [reply assets/schema_input.json:105]: We only need one tsv file which contains reads ID and taxid , for instance SRR13439790_pe_SRR13439790_diamond.diamond.tsv

**Review by sofstam — APPROVED** [review]

**Pull request merged**